### ISSUE TITLE
ImportError: cannot import name '_DictWithDeprecatedKeys' from 'sklearn.covariance._graph_lasso' on sklearn `main`

### ISSUE THREAD

**Issue by ogrisel** [skops-dev/skops#187]
It might makes sense to protect against `ImportError` when importing private classes / functions from scikit-learn (or numpy / scipy) to make it possible to use skops across versions of those libraries.

In particular at the moment, importing `skops.io` fails with:

```python-traceback
>>> from skops import io
Traceback (most recent call last):
  Input In [13] in <cell line: 1>
    from skops import io
  File ~/code/skops/skops/io/__init__.py:1 in <module>
    from ._persist import load, save
  File ~/code/skops/skops/io/_persist.py:19 in <module>
    module = importlib.import_module(module_name, package="skops.io")
  File ~/mambaforge/envs/dev/lib/python3.10/importlib/__init__.py:126 in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File ~/code/skops/skops/io/_sklearn.py:6 in <module>
    from sklearn.covariance._graph_lasso import _DictWithDeprecatedKeys
ImportError: cannot import name '_DictWithDeprecatedKeys' from 'sklearn.covariance._graph_lasso' (/Users/ogrisel/code/scikit-learn/sklearn/covariance/_graph_lasso.py)
```

**Comment by BenjaminBossan** [comment]
Good idea, thanks for the suggestion.

For this particular issue, IIUC from https://github.com/scikit-learn/scikit-learn/pull/24599, we can just replace it with a normal dict, right?

**Comment by BenjaminBossan** [comment]
Should be fixed now via #188
